dashboard/app/controllers/activities_controller.rb

```diff
@@ -163,7 +164,13 @@ def track_progress_for_user
       level_source_id: @level_source.try(:id)
     }
 
-    allow_activity_writes = Gatekeeper.allows('activities', where: {script_name: @script_level.script.name}, default: true)
+    is_multi_assessment = false
+    if @script_level
+      is_sublevel = !@script_level.levels.include?(@level)
+      is_multi_assessment = @script_level.assessment && @level.is_a?(Multi) && !is_sublevel
```

Comment:
why do we want `@level` to not be a sublevel here (i.e., why do we need `&& !is_sublevel` on this line)?

Reply:
great question. on a level_group a.k.a. multi-page assessment, the `@level` could be a multiple choice question while `@script_level` would be the level group, which is an assessment. within level groups, whenever you click a multiple choice answer we do a milestone post for it, so without this check, we would do way many writes to `activities` for multiple choice questions that we don't care about knowing the history for. there is a unit test which illustrates this scenario, but I could add a comment here too if that would be helpful.

Reply:
oh, i see. thank you! i do think a comment here would be useful, just to make this intent extra clear